[master]: unexpected "panic: unsupported call:" while processing unrecognised function

Previously (v0.10.0), this invalid query:
```
select foobar(fieldsCreate) from "shard"
```

would cause an error message to be displayed:

![v0.10.0 behaviour](https://www.dropbox.com/s/ls677av3qnvnbmb/Screenshot%202016-02-21%2021.27.49.png?dl=0&raw=1)

Currently, the same query causes this panic:

```
[query] 2016/02/21 21:28:46 SELECT foobar(fieldsCreate) FROM _internal.monitor."shard"
panic: unsupported call: foobar
goroutine 41 [running]:
github.com/influxdata/influxdb/influxql.buildExprIterator(0x19a5510, 0xc8206ae720, 0x19a5868, 0xc8202a2ca0, 0x19a5510, 0xc8206ae720, 0xd2d8d8, 0x0, 0x0, 0xc8202833b0, ...)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/influxql/select.go:324 +0x1204
github.com/influxdata/influxdb/influxql.buildFieldIterators.func1(0xc82007c040, 0x1, 0x1, 0x19a5868, 0xc8202a2ca0, 0xc82042d600, 0xc820283420, 0x1, 0x1, 0x0, ...)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/influxql/select.go:151 +0x5b6
github.com/influxdata/influxdb/influxql.buildFieldIterators(0xc82007c040, 0x1, 0x1, 0x19a5868, 0xc8202a2ca0, 0x0, 0x0, 0xd2d8d8, 0x0, 0x0, ...)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/influxql/select.go:137 +0xf8
github.com/influxdata/influxdb/influxql.Select(0xc82020a340, 0x19a5868, 0xc8202a2ca0, 0xc82042dcd0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/influxql/select.go:77 +0xba9
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeSelectStatement(0xc8201804c0, 0xc82020a340, 0x2710, 0x0, 0xc820020e40, 0xc820020de0, 0x0, 0x0)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/cluster/query_executor.go:429 +0x6ee
github.com/influxdata/influxdb/cluster.(*QueryExecutor).executeQuery(0xc8201804c0, 0xc8202a25a0, 0xc82027c03d, 0x9, 0x2710, 0xc820020de0, 0xc820020e40)
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/cluster/query_executor.go:108 +0x775
created by github.com/influxdata/influxdb/cluster.(*QueryExecutor).ExecuteQuery
	/Users/jonseymour/ninja/go/src/github.com/influxdata/influxdb/cluster/query_executor.go:61 +0x96
```

/cc @benbjohnson

--- a/influxql/select.go
+++ b/influxql/select.go
@@ -321,7 +321,7 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions) (Iter
 			opt.StartTime, opt.EndTime = MinTime, MaxTime
 			return newDerivativeIterator(input, opt, interval, isNonNegative), nil
 		default:
-			panic(fmt.Sprintf("unsupported call: %s", expr.Name))
+			return nil, fmt.Errorf("unsupported call: %s", expr.Name)
 		}
 
 		if err != nil {

--- a/influxql/select_test.go
+++ b/influxql/select_test.go
@@ -1584,3 +1584,15 @@ func TestSelect_Derivative_Integer(t *testing.T) {
 		t.Fatalf("unexpected points: %s", spew.Sdump(a))
 	}
 }
+
+func TestSelect_UnsupportedCall(t *testing.T) {
+	var ic IteratorCreator
+	ic.CreateIteratorFn = func(opt influxql.IteratorOptions) (influxql.Iterator, error) {
+		return &FloatIterator{}, nil
+	}
+
+	_, err := influxql.Select(MustParseSelectStatement(`SELECT foobar(value) FROM cpu`), &ic, nil)
+	if err == nil || err.Error() != "unsupported call: foobar" {
+		t.Errorf("unexpected error: %s", err)
+	}
+}

